src/pages/BookmarkPage.jsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from 'react';
+
+import BookmarkCard from '../components/BookmarkCard';
+import data from './mockData';
+
+const numberOfItemsOnClick = 4;
+let arrayHoldingWholeData = [];
+
+function BookmarkPage() {
+  const [next, setNext] = useState(8);
+  const [bookmarkedItemToShow, setBookmarkedItemToShow] = useState([]);
+
+  const loopThroughItem = (startIndex, endIndex) => {
+    const slicedIndex = data.slice(startIndex, endIndex);
+    arrayHoldingWholeData = [...arrayHoldingWholeData, ...slicedIndex];
+    setBookmarkedItemToShow(arrayHoldingWholeData);
+  };
+
+  useEffect(() => {
+    loopThroughItem(0, 8);
+  }, []);
+
+  const handleMoreItems = () => {
+    loopThroughItem(next, next + numberOfItemsOnClick);
+    setNext(next + numberOfItemsOnClick);
+  };
+
+  const removeBookmarked = (id) => {
+    const filteredBookmarks = bookmarkedItemToShow.filter((i) => i.id !== id);
+    setBookmarkedItemToShow(filteredBookmarks);
+  };
+
+  return data && data.length ? (
```

Comment:
Can you make this an early return, it looks weird like this because you need to scroll too far to see what the else if does.

Reply:
if (!data.length)
return No bookmarks